<?xml version="1.0"?>


<launch>

    <param name='sim_scale' type='int' value='1'/>


  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="extra_gazebo_args" default=""/>
  <arg name="gui" default="true"/>
  <arg name="recording" default="false"/>
  <arg name="world" default="house2" />
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="physics" default="ode"/>
  <arg name="verbose" default="false"/>
  <arg name="world_name" value="$(find exp3)/worlds/$(arg world).world"/>
  <arg name="respawn_gazebo" default="false"/>
  <arg name="use_clock_frequency" default="false"/>
  <arg name="pub_clock_frequency" default="100"/>


  <param name="/use_sim_time" value="$(arg use_sim_time)"/>


  <arg unless="$(arg paused)" name="command_arg1" value=""/>
  <arg     if="$(arg paused)" name="command_arg1" value="-u"/>
  <arg unless="$(arg recording)" name="command_arg2" value=""/>
  <arg     if="$(arg recording)" name="command_arg2" value="-r"/>
  <arg unless="$(arg verbose)" name="command_arg3" value=""/>
  <arg     if="$(arg verbose)" name="command_arg3" value="--verbose"/>
  <arg unless="$(arg debug)" name="script_type" value="gzserver"/>
  <arg     if="$(arg debug)" name="script_type" value="debug"/>

  <group if="$(arg use_clock_frequency)">
    <param name="gazebo/pub_clock_frequency" value="$(arg pub_clock_frequency)" />
  </group>
  <node name="gazebo" pkg="gazebo_ros" type="$(arg script_type)" respawn="$(arg respawn_gazebo)" output="screen"
	args="$(arg command_arg1) $(arg command_arg2) $(arg command_arg3) -e $(arg physics) $(arg extra_gazebo_args) $(arg world_name)" />
	

  <group if="$(arg gui)">
    <node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" respawn="false" output="screen"/>
  </group>
  
 
  <param name="use_gui" value="false"/>
   <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" ></node>
   <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" /> 
   <node name="rviz" pkg="rviz" type="rviz" args="-d $(find exp3)/config/config.rviz" />

  <param name="human_description1" textfile="$(find exp3)/urdf/human.urdf" />   
  <node name="urdf_spawner_human1" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
        args="-urdf -model human1 -param human_description1 -x -6 -y 8.5 -z 0.34 -R 0 -P 0 -Y -0.75"/>


    <param name='state' type='string' value='sleep'/>
    <param name='map/x_max' type='int' value='9'/>
    <param name='map/y_max' type='int' value='9'/>
    <param name='map/x_min' type='int' value='-9'/>
    <param name='map/y_min' type='int' value='-9'/>
    <param name='home/x' type='int' value='-5'/>
    <param name='home/y' type='int' value='8'/>
    <param name='blue/x' type='double' value='100'/>
    <param name='blue/y' type='double' value='100'/>
    <param name='red/x' type='double' value='100'/>
    <param name='red/y' type='double' value='100'/>
    <param name='green/x' type='double' value='100'/>
    <param name='green/y' type='double' value='100'/>
    <param name='yellow/x' type='double' value='100'/>
    <param name='yellow/y' type='double' value='100'/>
    <param name='magenta/x' type='double' value='100'/>
    <param name='magenta/y' type='double' value='100'/>
    <param name='black/x' type='double' value='100'/>
    <param name='black/y' type='double' value='100'/>
    <param name='new_ball_detected' type='int' value='0'/>
    <param name='unknown_ball' type='int' value='100'/>
    

    <param name='play_task_status' type='int' value='0'/>


    <rosparam param="room_list">['entrance', 'closet', 'livingroom',
                                 'kitchen', 'bathroom', 'bedroom']</rosparam>


    <param name="robot_description" 
	    command="$(find xacro)/xacro --inorder $(find exp3)/urdf/robot.xacro"/>


  	<node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
      respawn="false" output="screen"
	    args="-urdf -model robot1 -param robot_description -x -5 -y 8 -Y -1.57 -z 0.05"/>
    
    <node pkg='exp3' type='dog_fsm.py'
      name='dog_fsm_node' output='screen' respawn="false"/>

    <node pkg='exp3' type='dog_vision.py'
      name='dog_vision_node' output='screen' respawn="false"/>

    <node pkg='exp3' type='human.py'
      name='human_node' output='screen' respawn="false"/>


    <node pkg='exp3' type='ball_server.py'
      name='ball_server_node' output='screen' respawn="false"/>

    <node pkg="exp3" type="exp3_node" respawn="false" name="explore_node">
      <param name="robot_base_frame" value="link_chassis"/>
      <param name="costmap_topic" value="map"/>
      <param name="costmap_updates_topic" value="map_updates"/>
      <param name="visualize" value="true"/>
      <param name="planner_frequency" value="0.33"/>
      <param name="progress_timeout" value="30.0"/>
      <param name="potential_scale" value="3.0"/>
      <param name="orientation_scale" value="0.0"/>
      <param name="gain_scale" value="1.0"/>
      <param name="transform_tolerance" value="0.3"/>
      <param name="min_frontier_size" value="0.75"/>
    </node>


	<param name="use_sim_time" value="true"/>
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
      <param name="base_frame" value="link_chassis"/>
      <param name="map_update_interval" value="5.0"/>
      <param name="maxUrange" value="8.0"/>
      <param name="sigma" value="0.05"/>
      <param name="kernelSize" value="1"/>
      <param name="lstep" value="0.05"/>
      <param name="astep" value="0.05"/>
      <param name="iterations" value="5"/>
      <param name="lsigma" value="0.075"/>
      <param name="ogain" value="3.0"/>
      <param name="lskip" value="0"/>
      <param name="srr" value="0.1"/>
      <param name="srt" value="0.2"/>
      <param name="str" value="0.1"/>
      <param name="stt" value="0.2"/>
      <param name="linearUpdate" value="1.0"/>
      <param name="angularUpdate" value="0.5"/>
      <param name="temporalUpdate" value="3.0"/>
      <param name="resampleThreshold" value="0.5"/>
      <param name="particles" value="50"/>
      <param name="xmin" value="-50.0"/>
      <param name="ymin" value="-50.0"/>
      <param name="xmax" value="50.0"/>
      <param name="ymax" value="50.0"/>
      <param name="delta" value="0.05"/>
      <param name="llsamplerange" value="0.01"/>
      <param name="llsamplestep" value="0.01"/>
      <param name="lasamplerange" value="0.005"/>
      <param name="lasamplestep" value="0.005"/>
      <param name="transform_publish_period" value="0.0005"/>
    </node>



  <node pkg="move_base" type="move_base" respawn="false" name="move_base">

    <rosparam file="$(find exp3)/param/costmap_common_params.yaml" command="load" ns="global_costmap"/>
    <rosparam file="$(find exp3)/param/costmap_common_params.yaml" command="load" ns="local_costmap"/>
    <rosparam file="$(find exp3)/param/local_costmap_params.yaml" command="load"/>
    <rosparam file="$(find exp3)/param/global_costmap_params.yaml" command="load"/>
    <rosparam file="$(find exp3)/param/base_local_planner_params.yaml" command="load"/>
    <rosparam file="$(find exp3)/param/move_base_params.yaml" command="load"/>
  </node>
</launch>

